pkg/controller/controller.go

```diff
@@ -617,6 +620,104 @@ func convertCatalog(in *osb.CatalogResponse) ([]*v1beta1.ClusterServiceClass, []
 	return serviceClasses, servicePlans, nil
 }
 
+func convertAndFilterCatalog(in *osb.CatalogResponse, restrictions *v1beta1.CatalogRestrictions) ([]*v1beta1.ClusterServiceClass, []*v1beta1.ClusterServicePlan, error) {
```

Comment:
In the future, we could make this just work with `CommonService(Class|Plan)Spec`, and have wrappers that built the right types for cluster and ns-scoped. WDYT?

Reply:
this currently uses the parent service class name in part of the plan query to aid in the filtering. We are going to need a general strategy on how to deal with this all over the place in the codebase.

Reply:
also why I did not just copy/paste yet, I wanted to look into seeing if I can make it work as you asked.